monai/transforms/spatial/functional.py

```diff
@@ -0,0 +1,405 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A collection of "vanilla" transforms for spatial operations
+https://github.com/Project-MONAI/MONAI/wiki/MONAI_Design
+"""
+
+from __future__ import annotations
+
+import math
+import warnings
+from enum import Enum
+
+import numpy as np
+import torch
+
+import monai
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.data.meta_obj import get_track_meta
+from monai.data.meta_tensor import MetaTensor
+from monai.data.utils import AFFINE_TOL, compute_shape_offset, to_affine_nd
+from monai.networks.layers import AffineTransform
+from monai.networks.utils import normalize_transform
+from monai.transforms.croppad.array import ResizeWithPadOrCrop
+from monai.transforms.intensity.array import GaussianSmooth
+from monai.transforms.inverse import TraceableTransform
+from monai.transforms.utils import create_rotate, create_scale, create_translate, scale_affine
+from monai.transforms.utils_pytorch_numpy_unification import allclose
+from monai.utils import (
+    TraceKeys,
+    convert_data_type,
+    convert_to_dst_type,
+    convert_to_numpy,
+    convert_to_tensor,
+    ensure_tuple,
+    ensure_tuple_rep,
+    fall_back_tuple,
+    optional_import,
+)
+
+nib, has_nib = optional_import("nibabel")
+cupy, _ = optional_import("cupy")
+cupy_ndi, _ = optional_import("cupyx.scipy.ndimage")
+np_ndi, _ = optional_import("scipy.ndimage")
+
+__all__ = ["spatial_resample", "orientation", "flip", "resize", "rotate", "zoom", "rotate90", "affine_func"]
+
+
+def spatial_resample(
+    img, dst_affine, spatial_size, mode, padding_mode, align_corners, dtype_pt, transform_info
```

Comment:
missing doc strings. The original call function has them all thus I will add them in my PR